src/widgets/hierarchical-menu/hierarchical-menu.js

```diff
@@ -46,7 +46,7 @@ hierarchicalMenu({
   [ separator=' > ' ],
   [ rootPath ],
   [ showParentLevel=true ],
-  [ limit=1000 ],
+  [ limit=30 ],
```

Comment:
30 is a lot. I think 10 should be enough especially since we have the show more config.